In [5]:
import numpy as np
import statsmodels.api as sm

# Exercise 5.1

Given:

$$
\begin{align}
p(Y|X) &= \frac{e^{\beta_0 + \beta_1 X}}{1+e^{\beta_0 + \beta_1 X}}\\
p(X) &= p(Y=1|X)\\
\log \left( \frac{p(X)}{1-p(X)} \right) &= \beta_0 + \beta_1 X\\
\end{align}
$$

Then:

$$
\begin{align}
\log \left[ \frac{p(Y=1|X)}{1-p(Y=1|X)} \right] &= \beta_0 + \beta_1 X\\
\log \left[ \frac{\frac{e^{\beta_0 + \beta_1 X}}{1+e^{\beta_0 + \beta_1 X}}}{1 - \frac{e^{\beta_0 + \beta_1 X}}{1+e^{\beta_0 + \beta_1 X}}} \right] &= \beta_0 + \beta_1 X\\
\log\left[\frac{e^{\beta_{0} + \beta_{1}X}}{1+e^{\beta_{0} + \beta_{1}X}} \cdot \frac{1 + e^{\beta_{0} + \beta_{1}X}}{[1+e^{\beta_{0} + \beta_{1}X}] - e^{\beta_{0} + \beta_{1}X}} \right] &= \beta_{0} + \beta_{1}X\\
\log\left[e^{\beta_{0} + \beta_{1}X}\right] &= \beta_{0} + \beta_{1}X \quad \text{, assuming log = ln}\\
\beta_{0} + \beta_{1}X &= \beta_{0} + \beta_{1}X
\end{align}
$$

## "Log-odds" or "Logit"

$$
\underbrace{\ln\left(\frac{p(X)}{1 - p(X)}\right)}_{\text{"log-odds" or "logit"}} = \beta_0 + \beta_1 X
$$

# Exercise 5.2

$$
odds = \dfrac{p(X)}{1 - p(X)}
$$

### a)

$$
\begin{align}
0.47 &= \frac{p(x)}{1-p(x)}\\
0.47 - 0.47p(x) &= p(x)\\
0.47 &= 1.47 \cdot p(x)\\
\frac{0.47}{1.47} &= p(x)\\
0.3197 &\approx p(x)
\end{align}
$$

### b)

$$
\mathrm{odds} = \frac{0.25}{1 - 0.25} = \frac{0.25}{0.75} = \frac{1}{3} = 0.3\overline{3}
$$

# Exercise 5.3

Simple Logistic Regression Model:

$$
p(Y = 1|X) = \dfrac{e^{\beta_0 + \beta_1 X + ... + \beta_n X}}{1+e^{\beta_0 + \beta_1 X + ... + \beta_n X}}\\
$$

### a)

In [6]:
b0 = -6
b1 = 0.05
b2 = 1

x1 = 40
x2 = 4.5

y_pred = np.exp(b0 + b1 * x1 + b2 * x2) / (1 + np.exp(b0 + b1 * x1 + b2 * x2))
y_pred

np.float64(0.6224593312018546)

### b)

$$
\begin{align}
0.7 &= \frac{e^{-6+0.05 x_{1} + 1 \cdot 4.5}}{1 \cdot e^{-6 + 0.05x_{1}+1 \cdot 4.5}}\\
0.7 \cdot (1+e^{0.05 x_{1} - 1.5}) &= e^{0.05 x_{1}-1.5}\\
0.7 \cdot (1+e^{0.05 x_{1}} \cdot e^{-1.5}) &= e^{0.05 x_{1}} \cdot e^{-1.5}\\
0.7 + 0.7 e^{0.05 x_{1}} \cdot e^{-1.5} &= e^{0.05 x_{1}} \cdot e^{-1.5}\\
0.7 &= e^{0.05 x_{1}} \cdot e^{-1.5}-0.7 e^{0.05 x_{1}} \cdot e^{-1.5}\\
0.7 &= 0.3 e^{0.05 x_{1}} \cdot e^{-1.5}\\
\frac{0.7}{0.3} &= e^{0.05 x_{1}} \cdot e^{-1.5}\\
\frac{0.7}{0.3} \cdot \frac{1}{e^{-1.5}} &= e^{0.05 x_{1}}\\
\ln \left[\frac{0.7}{0.3 \cdot e^{-1.5}} \right] &= 0.05 x_{1}\\
\ln \left[\frac{0.7}{0.3 \cdot e^{-1.5}} \right] : 0.05 &= x_{1}\\
\end{align}
$$

In [15]:
x1 = np.log(0.7 / (0.3 * np.exp(-1.5))) / 0.05

x2 = 4.5
b0 = -6
b1 = 0.05
b2 = 1

y_pred = np.exp(b0 + b1 * x1 + b2 * x2) / (1 + np.exp(b0 + b1 * x1 + b2 * x2))
x1, y_pred

(np.float64(46.94595720774407), np.float64(0.7000000000000001))

To have a 70% chance of scoring a mark of 5.5 or better, we would expect the student from exercise a) to study at least 46.95 hours.